File: scala/src/main/scala/com/shekhargulati/ninetynine_problems/scala/_01_lists/P25.scala
package com.shekhargulati.ninetynine_problems.scala._01_lists

/**
  * (*) Generate a random permutation of the elements of a list.
  */
object P25 {

  def randomPermutation[T](list: List[T]): List[T] = P23.randomSelectR(list, list.size)
}

File: java8/src/main/java/com/shekhargulati/ninetynine_problems/_01_lists/P25.java
package com.shekhargulati.ninetynine_problems._01_lists;

import java.util.List;

/**
 * (*) Generate a random permutation of the elements of a list.
 */
public class P25 {

    public static <T> List<T> randomPermutation(List<T> list) {
        return P23.randomSelectR(list, list.size());
    }
}
